sematic/ui/src/pipelines/external_resource/ExternalResourceState.tsx

```diff
@@ -0,0 +1,64 @@
+import TimelineConnector from "@mui/lab/TimelineConnector/TimelineConnector";
+import TimelineContent from "@mui/lab/TimelineContent/TimelineContent";
+import TimelineDot, { TimelineDotProps } from "@mui/lab/TimelineDot/TimelineDot";
+import TimelineItem from "@mui/lab/TimelineItem/TimelineItem";
+import TimelineSeparator from "@mui/lab/TimelineSeparator/TimelineSeparator";
+import { TERMINATE_STATE } from "../../hooks/externalResourceHooks";
+import { format } from 'date-fns'
+import { useMemo } from "react";
+import Typography from "@mui/material/Typography/Typography";
+import { ExternalResourceHistorySerialization, ExternalResourceState as ExternalResourceStateType  } from "../../Models";
+import { styled } from "@mui/system";
+
+const HighLightenedText = styled(Typography)`
+    font-weight: bold;
+    display: inline;
+`;
+
+const SpacedText = styled(Typography)`
+    margin: 4px 0;
+`;
+
+interface ExternalResourceStateProps {
+    historyRecord: ExternalResourceHistorySerialization;
+    isLast: boolean;
+}
+
+function TimelineDotWithColor({resourceState} : {resourceState: ExternalResourceStateType}) {
+    let color: TimelineDotProps["color"] = 'grey';
+    if (["ACTIVATING", "DEACTIVATING"].includes(resourceState)) {
+        color='primary';
+    }
+    if ('ACTIVE' === resourceState) {
+        color='success';
+    }
+    if (["CREATED", "DEACTIVATED"].includes(resourceState)) {
+        color='grey';
+    }
+    return <TimelineDot color={color}/>;
+};
+
+export default function ExternalResourceState({historyRecord, isLast}: ExternalResourceStateProps) {
+    const {root_type, values: {id, status: {values: { state, last_update_epoch_time, message }}} } = historyRecord;
+    const timeString = useMemo(
+        () => format(
+            new Date(last_update_epoch_time * 1000), 
+            'LLL\xa0d,\xa0yyyy\xa0h:mm:ss\xa0a'), [last_update_epoch_time]);
+
+    const resourceName = root_type.type[1];
+
+    return <TimelineItem key={id}>
+        <TimelineSeparator>
+            {isLast ? 
+                <TimelineDotWithColor resourceState={state} /> 
+                : <TimelineDot />}
+            {state !== TERMINATE_STATE && <TimelineConnector />}
+        </TimelineSeparator>
+        <TimelineContent>
+            {`[${state}] `} 
```

Comment:
Can you remove the square bracket and put it in a [Chip](https://mui.com/material-ui/react-chip/)?

Reply:
Sure. It now looks like this:

<img width="561" alt="image" src="https://user-images.githubusercontent.com/1046489/214732544-be0b0729-80b4-47a8-bf22-dec1fbdce0a8.png">

The chip color will be in sync with the color of the dot --- in that, only the last entry will be colorized.